source/slang/slang-check-expr.cpp

```diff
@@ -2371,19 +2384,99 @@ namespace Slang
         return result;
     }
 
+    Expr* SemanticsExprVisitor::convertToLogicOperatorExpr(InvokeExpr* expr, bool& isArgumentsChecked)
+    {
+        LogicOperatorShortCircuitExpr* newExpr = nullptr;
+        isArgumentsChecked = false;
+
+        // If the logic expression is inside the generic parameter list, it cannot support short-circuit
+        // which will generate the ifelse branch.
+        if (!m_shouldShortCircuitLogicExpr)
+        {
+            return nullptr;
+        }
+
+        const ReflectClassInfo& classInfo = expr->functionExpr->getClassInfo();
```

Comment:
`expr->functionExpr` might be nullptr. Always check for nullptr in semantics visitor.
Also it should be `auto astType = expr->functionExpr->astNodeType;`.

Reply:
Usually we don't directly compare astType, and we just write:
`if (as<VarExpr>(expr->functionExpr)`
This automatically takes care of the case of expr->functionExpr being null.

Reply:
Thanks for the tips, updated.